platform/engineering/design-docs/2020-09-24-package-forms-library.md

```diff
@@ -0,0 +1,217 @@
+
+Finally, there are some testing helpers that live in the `src/applications/hca` directory and are used in forms library tests:
+
+- [`01-sip-autosave.e2e.spec.js`](https://github.com/department-of-veterans-affairs/vets-website/blob/cc4a3172edc242bf60c93aec5b170c734dc57985/src/platform/forms/tests/save-in-progress/01-sip-autosave.e2e.spec.js#L4)
+- [`01-sip-fail-load.e2e.spec.js`](https://github.com/department-of-veterans-affairs/vets-website/blob/cc4a3172edc242bf60c93aec5b170c734dc57985/src/platform/forms/tests/save-in-progress/01-sip-load-fail.e2e.spec.js#L3)
+- [`01-sip-finish-later.e2e.spec.js`](https://github.com/department-of-veterans-affairs/vets-website/blob/cc4a3172edc242bf60c93aec5b170c734dc57985/src/platform/forms/tests/save-in-progress/01-sip-finish-later.e2e.spec.js#L4)
+- [`01-sip-review.e2e.spec.js`](https://github.com/department-of-veterans-affairs/vets-website/blob/cc4a3172edc242bf60c93aec5b170c734dc57985/src/platform/forms/tests/save-in-progress/01-sip-review.e2e.spec.js#L3)
+
+
+#### Platform entanglement
+
+In addition to the forms library, the platform provides other code that is used both in the forms library and in applications. 
+
+- `utilities`
+- `monitoring`
+- `site-wide`
+- `user`
+- `static-data`
+
+These aren't _strictly_ part of the forms library, but since the forms library makes use of its parent platform code, it becomes more difficult to place clear boundaries around the forms library.
+
+### High Level Design
+
+The forms library will be consolidated into a single place and bundled as a Node package.
+This will allow VFS teams to import it as if it were any other Node module.
+
+This change should not have a noticeable functional impact on individual app teams.
+Towards the end the only change that should be made on the app side of things is changing where forms library code/styles are imported from.
+
+<!-- The source for this diagram is the `*.odg` file in the same directory, a LibreOffice Draw doc-->
+![Forms library bundling](images/forms-library-package/forms-library-package.svg)
+
+## Specifics
+
+### Detailed Design
+
+A few things will have to happen in order pave the way for additional changes that will bundle the forms library.
+
+#### Preparation
+
+- The app-specific code mentioned in [Application Entanglement](#application-entanglement) will have to be moved so that it no longer lives in the forms library.
+- Use the `no-unresolved-modules` ESLint plugin to [find & remove unused code](https://github.com/department-of-veterans-affairs/va.gov-team/issues/8763) in the `forms` and `forms-library` directories
+
+#### Beginning the bundling
+
+- Copy what remains from `forms` into `forms-system`
+- Create `platform/packages/forms-library` and move the contents of `platform/forms-system` into it
+- Add a `package.json` file to the new `forms-library` package directory
```

Comment:
What will go in this `package.json`? I'm most interested in dependencies and how they will impact the build process. For example, if I `yarn install` from the root, will it install the `forms-library` dependencies?

Reply:
I'm happy to report that a `yarn install` at the root of `vets-website` does install dependencies in the child `package.json`

Reply:
I just tried this myself:
1) `cd src/platform/forms`
2) `npm init`
3) `yarn add chalk` (just to add a dependency)
4) `rm -rf node_modules`
5) `cd ../../../` (back to `vets-website` root)
6) `yarn`
7) `ls src/platform/forms`

No `node_modules` directory was present as sibling to the `package.json` I added in the sub-directory `src/platform/forms/`. :disappointed: